connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java

```diff
@@ -0,0 +1,278 @@
+                + "but was rejected with an error response for some reason.");
+            System.out.println("Error Message:" + oe.getErrorMessage());
+            System.out.println("Error Code:" + oe.getErrorCode());
+            System.out.println("Request ID:" + oe.getRequestId());
+            System.out.println("Host ID:" + oe.getHostId());
+        } catch (ClientException ce) {
+            System.out.println("Caught an ClientException, which means the client encountered "
+                    + "a serious internal problem while trying to communicate with OSS, "
+                    + "such as not being able to access the network.");
+            System.out.println("Error Message:" + ce.getMessage());
+        }
+    }
+
+    private void handleRecord(ConnectRecord record) throws ConnectException, IOException {
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("data", record.getData());
+        String context = JSON.toJSONString(jsonObject);
+        String prefix = genFilePrefixByPartition(record);
+        if (enableBatchPut) {
+            if (!recordMap.containsKey(prefix)) {
+                recordMap.put(prefix, new ArrayList<>()); 
+            }
+            recordMap.get(prefix).add(context);
+        } else {
+            String absolutePath = prefix + objectName;
+            long appendOffset = genObjectOffset(record, absolutePath);
+            putOss(absolutePath, appendOffset, context);
+            lastTimeStamp = record.getTimestamp();
+        }
+    }
+
+    @Override
+    public void put(List<ConnectRecord> sinkRecords) throws ConnectException {
+        try {
+            sinkRecords.forEach(sinkRecord -> {
+                try {
+                    handleRecord(sinkRecord);
+                } catch (OSSException oe) {
+                    System.out.println("Caught an OSSException, which means your request made it to OSS, "
+                            + "but was rejected with an error response for some reason.");
+                    System.out.println("Error Message:" + oe.getErrorMessage());
+                    System.out.println("Error Code:" + oe.getErrorCode());
+                    System.out.println("Request ID:" + oe.getRequestId());
+                    System.out.println("Host ID:" + oe.getHostId());
+                } catch (ClientException ce) {
+                    System.out.println("Caught an ClientException, which means the client encountered "
+                            + "a serious internal problem while trying to communicate with OSS, "
+                            + "such as not being able to access the network.");
+                    System.out.println("Error Message:" + ce.getMessage());
+                } catch (Exception e) {
+                    log.error("OSSSinkTask | genObjectOffset | error => ", e);
+                }
+            });
+            if (enableBatchPut && !recordMap.isEmpty()) {  
```

Comment:
When any exception happens above, I think processMap() should not be called.

Reply:
Added a boolean variable `hasException` to ensure `processMap` is called only if all records are processed successfully.